zcash_primitives/src/transaction/tests.rs

```diff
@@ -72,11 +74,14 @@ fn tx_write_rejects_unexpected_binding_sig() {
 
 proptest! {
     #[test]
-    fn tx_serialization_roundtrip(tx in arb_branch_id().prop_flat_map(arb_tx)) {
+    fn tx_serialization_roundtrip(
+        (branch_id, tx) in arb_branch_id().prop_flat_map(|branch_id|
+            arb_tx(branch_id).prop_map(move |tx| (branch_id, tx))
+        )) {
         let mut txn_bytes = vec![];
         tx.write(&mut txn_bytes).unwrap();
 
-        let txo = Transaction::read(&txn_bytes[..]).unwrap();
+        let txo = Transaction::read(&txn_bytes[..], branch_id).unwrap();
```

Comment:
None of the changes in this file are necessary once the bug I identified is fixed, given that we aren't storing the consensus branch ID inside `Transaction` in this legacy branch.